src/sdg/producten/models/product.py

```diff
@@ -79,6 +84,48 @@ def upn_uri(self):
     def upn_label(self):
         return self.upn.upn_label
 
+    @property
+    def is_sdg_product(self):
+        return bool(self.upn.sdg)
+
+    @cached_property
+    def has_referentie_product_publicatie_datum(self):
+        try:
+            return bool(
+                Product.objects.get(
+                    catalogus__is_referentie_catalogus=True, generiek_product=self
+                ).active_version
+            )
+        except Product.DoesNotExist:
+            return None
+
+    @cached_property
+    def is_localized_generieke_tekst_empty(self):
+        localized_products = LocalizedGeneriekProduct.objects.filter(
+            generiek_product=self
+        ).values_list("generieke_tekst", flat=True)
+
+        return not all(localized_products)
```

Comment:
Since this functions are ONLY used for the product status: I prefer to move them inside the `product_status` function. These functions are not really pretty so I don't want to use this in the wild (read in the public Python interface). Once inside the `product_status` they also don't need the decorator anymore and can be "cached" as variable inside the function.